From be4c2a82fc522c7c33c589e270b1e2c47aa3f2db Mon Sep 17 00:00:00 2001
From: b02247 <b02247@freescale.com>
Date: Fri, 25 Jan 2013 16:20:50 +0800
Subject: [PATCH 10/12] ENGR00241758 Fix noise when input and output opened in
 same time.

When open input and output in the same time, and alsa-driver is in
sleep, opening input will cause output underrun and have noise.
Enlarge the startthreshold to reduce noise in this case.

Signed-off-by: b02247 <b02247@freescale.com>
---
 pcm.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pcm.c b/pcm.c
index a2d8166..a79da1d 100644
--- a/pcm.c
+++ b/pcm.c
@@ -1,7 +1,7 @@
 /* pcm.c
 **
 ** Copyright 2011, The Android Open Source Project
-** Copyright (C) 2012 Freescale Semiconductor, Inc.
+** Copyright (C) 2012-2013 Freescale Semiconductor, Inc.
 **
 ** Redistribution and use in source and binary forms, with or without
 ** modification, are permitted provided that the following conditions are met:
@@ -745,7 +745,7 @@ struct pcm *pcm_open(unsigned int card, unsigned int device,
             pcm->config.start_threshold = sparams.start_threshold = 1;
         else
             pcm->config.start_threshold = sparams.start_threshold =
-                config->period_count * config->period_size / 2;
+                config->period_count * config->period_size;
     } else
         sparams.start_threshold = config->start_threshold;
 
-- 
1.8.0

